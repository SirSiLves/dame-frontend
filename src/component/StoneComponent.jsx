import React, {Component} from 'react';
import crown from '../crown.png';
import '../stylesheet/stone.css';


class StoneComponent extends Component {

    render() {
        return (
            <>
                {this.getDiv()}
            </>
        )
    }

    //creates a empty div if no stone is there or place the stone on it
    getDiv() {
        if (this.props.stone == null) {
            return <div/>
        } else {
            return <div
                className={this.getStyles()}
                onClick={this.props.onClick}
            >
                {this.isQueen()}
            </div>
        }
    }

    //creates the styles for a stone if a stone is on the field
    getStyles() {
        let styles = "stone ";
        if (this.props.stone == null) {

        } else {
            //If there is a Stone, do a lot of stuff
            styles += this.props.stone.stoneColor.toLowerCase();

            styles += this.props.stone.alive ? "" : " hidden";
        }
        return styles;
    }

    //if the stone is a queen, add the crown image
    isQueen() {
        if (this.props.stone.queen) {
            return <img src={crown} className="queen" alt={"This is a queen"} height={40} width={40}/>
        } else {

        }
    }
}

export default StoneComponent
