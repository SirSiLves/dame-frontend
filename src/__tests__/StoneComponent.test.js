import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import StoneComponent from "../component/StoneComponent";

it('renders StoneComponent without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<StoneComponent />, div);
});

describe('StoneComponent Component', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<StoneComponent stone={stone} debug />);

        expect(component).toMatchSnapshot();
    });
});

const stone = {stoneId: 0, stoneColor: "BLACK", queen: false, alive: true};