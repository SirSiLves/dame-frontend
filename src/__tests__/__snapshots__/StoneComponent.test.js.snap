// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoneComponent Component should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
